design/sketch.webidl

```diff
@@ -505,7 +504,9 @@ interface WebGPURenderPassEncoder : WebGPUProgrammablePassEncoder {
     void setScissorRect(u32 x, u32 y, u32 width, u32 height);
 
     void setIndexBuffer(WebGPUBuffer buffer, u32 offset);
-    void setVertexBuffers(u32 startSlot, sequence<WebGPUBuffer> buffers, sequence<u32> offsets);
+    // "startSlot" refers to an index into the "inputs" sequence in the currently bound pipeline's WebGPUInputStateDescriptor.
+    // "buffers" and "offsets" are parallel.
+    void setVertexBuffers(u32 startSlot, sequence<WebGPUBuffer?> buffers, sequence<u32?> offsets);
```

Comment:
What's the reasoning for allowing holes in `setVertexBuffers`? It seems that it could be trivially replaced by multiple calls that don't have holes.

Reply:
Have we figured out whether holes in `setVertexBuffers` unbind the buffer resource, or leave the old resource bound? If we leave the old resource bound, if I destroy the buffer, will the validation layer complain? How can I explicitly unbind the resource?

My recommendation, which is a bit bizarre, is that a hole/undefined means "leave existing resource", explicit null means "undefined", and we remove the `startSlot` parameter from setVertexBuffers, meaning that someone can use the holes for the same purpose. Holes are more flexible.

Reply:
separating undefined and `null` seems like a recipe for troubles in WASM land

Reply:
Or I suppose we could do:

```webidl
enum WebGPUSetVertexBuffersEntry { "unset", "keep" }
partial interface WebGPURenderPassEncoder {
    void setVertexBuffers(u32 startSlot, sequence<WebGPUBuffer | WebGPUSetVertexBuffersEntry> buffers, sequence<u32?> offsets);
}
```

Reply:
Maybe in that case it would be better `startSlot` entirely and expect the full range to be passed? Otherwise we could accept ranges for keep/clear to reduce the amount repeated values.